src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -1190,40 +1194,64 @@ public void UpdateBubbleContent()
         {
             if (DynamoViewModel == null) return;
 
-            bool hasErrorOrWarning = NodeModel.IsInErrorState || NodeModel.State == ElementState.Warning || NodeModel.State == ElementState.PersistentWarning;
+            bool hasErrorOrWarning = NodeModel.IsInErrorState || NodeModel.State == ElementState.Warning;
+
+            // Persistent warnings should not be dismissed even if nodes are not involved in an execution as they can include:
+            // 1. Compile-time warnings in CBNs
+            // 2. Obsolete nodes with warnings
+            // 3. Dummy or unresolved nodes
+            if (NodeModel.State != ElementState.PersistentWarning && !NodeModel.IsInErrorState)
```

Comment:
This condition is now consistent with the one [above](https://github.com/DynamoDS/Dynamo/pull/12682/files#diff-97f7c391a3f292149633414ed4ac50ce1a73a1439bcc3307d20da0fb043a6d78R836).